### PR TITLE
Kick off 2.0-beta

### PR DESCRIPTION
**Pull request by mixonic**
Per https://github.com/Addepar/ember-table/issues/819 we're pursuing a 3.0 release for ember-table which drops IE11 support and Ember pre-2.8. As ember-table is likely the most active contributor of this library, I would also like to release a 2.0 of vertical collection dropping support for IE11 and Ember < 2.8.

This branch should kick that off.

### CODE REVIEW

**Comment by mixonic** [comment]
Annoyingly although ember-table supports back to Ember 2.4 (back to 1.11 on master!) this library is only tested in Ember 3.x and does not pass on 2.x.

**Comment by rwwagner90** [comment]
This seems fine so far. If we're dropping IE11, maybe we should also drop Ember 2.x?

**Comment by mixonic** [comment]
Addepar relies on 2.x support both for ember-table and various internal component implementations. Ember-table itself is probably the single largest consumer of this library, and since we can't drop Ember 2.x from ember-table yet we should not drop it from here. In fact I would like to *add* it to the test suite.

Quickly brainstorming, the advantage to dropping 2.x afaict would be a more stable component manager API which may allow an implementation here to operate at a lower level, or at least at a low level on a stable, public API instead of intimate API. But we could probably also use those APIs progressively during a vertical-collection 2.x cycle and drop weight at 3.x.

**Comment by apellerano-pw** [comment]
Is there any plans to integrate the horizontal-collection / virtual-collection pattern proposed in #269 into the next major release?

**Comment by knownasilya** [comment]
I'd love to at least get a Ember 3 and IE11 release, since Ember still supports IE11.

**Comment by rwwagner90** [comment]
I don't really know much about the internals of this repo, but happy to add anyone as maintainers if they want to help get this over the finish line!

**Comment by mixonic** [comment]
In alignment with a proposal at https://github.com/Addepar/ember-table/issues/819#issuecomment-772814116, I've opened a PR at https://github.com/html-next/vertical-collection/pull/325 moving the minimum supported version of Ember for the 2.0 branch to 2.8.

**Comment by mixonic** [comment]
@twokul and I discussed and we think the last item here is the `ember-raf-scheduler` dependency getting a bump.

**Comment by twokul** [comment]
As per discussion, https://github.com/html-next/vertical-collection/pull/328. I believe after it is merged, we can safely merge this PR and proceed with the release of 2.0.

**Review by twokul — APPROVED** [review]

File: .travis.yml
Comment (rwwagner90) [inline .travis.yml:32]: Why are we using both Travis and GitHub Actions? Shouldn't we use just one or the other? My preference would be GitHub Actions.

File: tests/dummy/config/targets.js
Comment (rwwagner90) [inline tests/dummy/config/targets.js:10]: Why remove this code?

**Review by rwwagner90 — CHANGES_REQUESTED** [review]

File: .travis.yml
Comment (mixonic) [reply .travis.yml:32]: Oh yeah this file should just get deleted. I thought I had already done it, but I've been bouncing around so many repos maybe I forgot.

File: tests/dummy/config/targets.js
Comment (mixonic) [reply tests/dummy/config/targets.js:10]: There is no need to build the dummy app with IE9 or any other legacy browser support for testing.

File: .travis.yml
Comment (twokul) [reply .travis.yml:32]: Indeed. We use GH Actions as a "regular" CI and Travis to catch the possible issues with "floating" dependencies. I agree we should standardise on one; I would not consider it blocking 2.0. Open to discussing of course! 😄

File: tests/dummy/config/targets.js
Comment (twokul) [reply tests/dummy/config/targets.js:10]: I do not think it is used anymore? 🤔

Comment (rwwagner90) [reply tests/dummy/config/targets.js:10]: This just comes from ember-cli blueprints though, right? If so, I would like to keep as much in line with the blueprints as we can.

File: .travis.yml
Comment (rwwagner90) [reply .travis.yml:32]: I think let's delete Travis for now and go with GH actions. I've got a floating dependencies job setup here if we want to copy that https://github.com/shipshapecode/ember-shepherd/blob/master/.github/workflows/main.yml#L25

Comment (twokul) [reply .travis.yml:32]: https://github.com/html-next/vertical-collection/pull/329

File: tests/dummy/config/targets.js
Comment (twokul) [reply tests/dummy/config/targets.js:10]: I am a little confused. We do not want to support IE, afaik. If so, that makes sense to remove the code. Am I missing something? Otherwise, both `isCI` and `isProduction` will be unused 🤔

Comment (twokul) [reply tests/dummy/config/targets.js:10]: https://github.com/html-next/vertical-collection/pull/330

Comment (twokul) [reply tests/dummy/config/targets.js:10]: Deleting it breaks the tests. Sorry, I do not see a good reason to keep this code around. I am happy to follow up with any change later. I do not think we should block the release on this 😄

Comment (rwwagner90) [reply tests/dummy/config/targets.js:10]: @twokul can we please just keep it as it was before this PR? It matched the ember-cli blueprints before.

Comment (twokul) [reply tests/dummy/config/targets.js]: I do not understand the reasoning but I have reverted the change as you asked https://github.com/html-next/vertical-collection/pull/321/commits/b0d44dbf9e40d721507321ab124bb5f590947905

Comment (twokul) [reply tests/dummy/config/targets.js]: The tests won't pass since `ember-maybe-import-regenerator` is not a dependency anymore

Comment (rwwagner90) [reply tests/dummy/config/targets.js:10]: The reasoning is to match the blueprints for ease of future Ember updates and better diffs. I do not disagree with you that the code is not necessary, I just always want to strive to match the current blueprints in all cases, unless we have a reason to diverge.

File: .github/workflows/ci.yml
Comment (rwwagner90) [inline .github/workflows/ci.yml:45]: Should we uncomment these, if we intend to support 2.8+?

Comment (twokul) [reply .github/workflows/ci.yml:45]: Now that `.travis.yml` is removed, I think so 🤔

File: tests/dummy/config/targets.js
Comment (mixonic) [reply tests/dummy/config/targets.js:10]: @rwwagner90 we have a reason to diverge, IE11 is not supported. Right?

File: .github/workflows/ci.yml
Comment (mixonic) [reply .github/workflows/ci.yml]: There is a comment right above them. The test suite doesn't run in Ember 2.x. The commented lines and comment above them explain that these are disabled due to that flaw.

We should get the tests to pass on 2.x, but I'm don't think we should block it since the CI suite today on master is busted generally.

Comment (rwwagner90) [reply .github/workflows/ci.yml]: It looks like we need to set ember-compatibility-helpers to 1.2.1 https://github.com/pzuraq/ember-compatibility-helpers/issues/47

Comment (mixonic) [reply .github/workflows/ci.yml]: This PR should not be blocked on fixing the 2.x test suite. I'm happy to entertain PRs against the beta or against master after this lands.

The code has copious coverage by ember-table's suite (which is running against 2.x) an against a massive codebase internally at Addepar. If is far more important that we get master into a functional state and get a release than it is to golf out the final legs of test suite repair.

File: tests/dummy/config/optional-features.json
Comment (mixonic) [inline tests/dummy/config/optional-features.json:2]: Why would you *enable* jquery?

Comment (rwwagner90) [reply tests/dummy/config/optional-features.json:2]: It's required for Ember 2.x

**Review by rwwagner90 — APPROVED** [review]

File: package.json
Comment (rwwagner90) [inline package.json:3]: Should we label this as a beta?

Comment (mixonic) [reply package.json:3]: This format of version number is commonly used by NPM packages to identify a pre-release. The published package on NPM is tagged `next`, which is basically equivalent to `beta` (only the `latest` tag on npm is for stable packages).

You prefer the `-beta` style versioning? Or were you referring to the npm tag?

**Comment by mixonic** [comment]
I've integrated this into ember-table and into a large codebase. I think we're good to land it and cut 2.0!

File: package.json
Comment (rwwagner90) [reply package.json:3]: I'm just used to seeing `-beta` on betas.

**Pull request merged**
